components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/PolicyPersistenceManagerTest.java

```diff
@@ -0,0 +1,330 @@
+        assertEquals(policy3FromDb.getPolicy(), inputPDPPolicy3.getPolicy());
+        assertEquals(policy3FromDb.getPolicyId(), inputPDPPolicy3.getPolicyId());
+        assertEquals(policyPersistenceManager.getPolicyOrder(inputPDPPolicy3.getPolicyId()),
+                inputPDPPolicy3.getPolicyOrder());
+        assertNull(policyPersistenceManager.getReferencedPolicy(inputPDPPolicy3.getPolicyId()));
+
+        // Verify the number of published policies.
+        List<String> policyIds = policyPersistenceManager.listPublishedPolicyIds();
+        assertEquals(policyIds.size(), 3);
+
+        // Verify the number of ordered policy identifiers.
+        String[] orderedPolicyIdentifiers = policyPersistenceManager.getOrderedPolicyIdentifiers();
+        assertEquals(orderedPolicyIdentifiers.length, 3);
+
+        // Verify the number of active policies.
+        String[] activePolicies = policyPersistenceManager.getActivePolicies();
+        assertEquals(activePolicies.length, 2);
+
+        policyPersistenceManager.deletePolicy(inputPDPPolicy1.getPolicyId());
+        policyPersistenceManager.deletePolicy(inputPDPPolicy2.getPolicyId());
+        policyPersistenceManager.deletePolicy(inputPDPPolicy3.getPolicyId());
+        assertFalse(policyPersistenceManager.isPolicyExist(inputPDPPolicy1.getPolicyId()));
+        assertFalse(policyPersistenceManager.isPolicyExist(inputPDPPolicy2.getPolicyId()));
+        assertFalse(policyPersistenceManager.isPolicyExist(inputPDPPolicy3.getPolicyId()));
+
+        policyPersistenceManager.removePolicy(inputPDPPolicy1.getPolicyId());
+        policyPersistenceManager.removePolicy(inputPDPPolicy2.getPolicyId());
+        policyPersistenceManager.removePolicy(inputPDPPolicy3.getPolicyId());
+    }
+
+    @DataProvider
+    public Object[][] updatePolicyStoreData() {
+
+        PolicyDTO papPolicy1 = new PolicyDTO();
+        papPolicy1.setPolicy(policyString1);
+        papPolicy1.setPolicyId("sample_policy1");
+
+        PolicyStoreDTO pdpPolicy1 = new PolicyStoreDTO();
+        pdpPolicy1.setPolicy(policyString1);
+        pdpPolicy1.setPolicyId("sample_policy1");
+        pdpPolicy1.setVersion("1");
+        pdpPolicy1.setActive(true);
+        pdpPolicy1.setSetActive(true);
+
+        PolicyStoreDTO pdpPolicy2 = new PolicyStoreDTO();
+        pdpPolicy2.setPolicyId("sample_policy1");
+        pdpPolicy2.setPolicyOrder(2);
+        pdpPolicy2.setSetOrder(true);
+
+        PolicyStoreDTO pdpPolicy3 = new PolicyStoreDTO();
+        pdpPolicy3.setPolicyId("sample_policy1");
+        pdpPolicy3.setActive(false);
+        pdpPolicy3.setSetActive(true);
+
+        return new Object[][]{
+                {papPolicy1, pdpPolicy1, pdpPolicy2, pdpPolicy3},
+        };
+    }
+
+    @Test(dataProvider = "updatePolicyStoreData", dependsOnMethods = {"testAddPDPPolicy"})
+    public void testUpdatePDPPolicy(Object papPolicy1, Object pdpPolicy1, Object pdpPolicy2, Object pdpPolicy3)
```

Comment:
Use proper object types types for params

Reply:
Refactored the code in the latest changes